torch/testing/_internal/common_methods_invocations.py

```diff
@@ -239,6 +240,17 @@ def __init__(self,
         self.aten_name = aten_name if aten_name is not None else name
         self.variant_test_name = variant_test_name
 
+        # Attribute to verify dynamic_dtypes are used.
+        self.dynamic_dtypes = any(map(lambda dtypes: isinstance(
+            dtypes, opinfo_helper._dynamic_dispatch_dtypes), (dtypes, dtypesIfCUDA)))
+
+        if self.dynamic_dtypes:
+            # Make sure `dtyesIfCUDA` is dynamic, if dynamic dispatch is used for CPU
+            # This is because, below we set dtypesIfCUDA to dtypes if they are None.
+            assert isinstance(dtypesIfCUDA, opinfo_helper._dynamic_dispatch_dtypes), \
```

Comment:
I'm still a little confused why we have to require this?

We can detect if someone set the dynamic dtypes using your class trick, and that's enough to test that people don't submit dynamic dtypes to CI. Does it really matter if people are using dynamic dtypes for, like, dtypesIfROCm only? Or dtypes but not dtypesIfCUDA?

Reply:
As we discussed offline, we would like the user to pass `dtypes` and `dtypesIfCUDA`, because if `dtypesIfCUDA` is not passed, `dtypesIfCUDA` is set to be `dtypes`. However for many operators CPU and CUDA dtypes are not same (support for float16, bfloat16 is not consistent) which will lead to failures and incorrect predictions.

So we want to keep this behaviour.